torch/csrc/jit/mobile/nnc/context.cpp

```diff
@@ -224,26 +233,39 @@ c10::impl::GenericList Function::run(
 
   // Fill in input tensors.
   TORCH_CHECK(
-      input_specs_.size() == inputs.size(),
+      input_specs_.size() == (inputs.size() + sym_shape_positions_.size()),
       "Input size doesn't match the spec, expect: ",
       input_specs_.size(),
       " actual: ",
       inputs.size());
+  std::vector<int64_t> scalar_values;
+  int offset = 0;
   for (const auto i : c10::irange(inputs.size())) {
     const c10::IValue& input = inputs[i];
+    const auto& spec = input_specs_[i];
     const auto& input_tensor = input.toTensor();
     TORCH_CHECK(
         input_specs_[i].validate(input_tensor), "Invalid input at pos: ", i);
     args[i] = input_tensor.data_ptr();
   }
+  offset += inputs.size();
+
+  for (const auto i : c10::irange(sym_shape_positions_.size())) {
+    const auto& sym_shape_pos = sym_shape_positions_[i];
+    const c10::IValue& input_tensor = inputs[sym_shape_pos.input_idx_];
+    auto dim = input_tensor.toTensor().size(sym_shape_pos.dim_idx_);
+    scalar_values.push_back(dim);
```

Comment:
Can we add `scalar_values.reserve(sym_shape_positions_.size())` before this loop, since we know the expected size?